src/main/scala/org/bitcoins/core/protocol/script/ScriptSignature.scala

```diff
@@ -39,15 +39,21 @@ sealed trait ScriptSignature extends NetworkElement with BitcoinSLogger {
 
 
   /**
-    * Derives the hash type for a given digitalSignature
+    * Derives the hash types for every digitalSignature in the ScriptSignature object
     *
-    * @param digitalSignature
     * @return
     */
-  def hashType(digitalSignature: ECDigitalSignature) : HashType = {
-    digitalSignature match {
-      case EmptyDigitalSignature => SIGHASH_ALL(Int32.one)
-      case sig : ECDigitalSignature => HashType(Seq(digitalSignature.bytes.last))
+  def hashTypes : Seq[HashType] = {
```

Comment:
Not sure if this is useful for anything, perhaps we should delete this function `hashType` and `hashTypes` all together? Would this cause any errors in our codebase?